datafusion/physical-expr/src/intervals/interval_aritmetic.rs

```diff
@@ -1079,6 +1089,117 @@ mod tests {
         Interval::make(lower, upper, (false, false))
     }
 
+    #[test]
+    fn boolean_interval_test() -> Result<()> {
+        // Demonstrate / document how boolean Intervals work
+        #[derive(Debug)]
+        struct TestCase {
+            lower: bool,
+            upper: bool,
+            /// expected bounds when lower is open, upper is open
+            expected_open_open: (bool, bool),
+            /// expected bounds when lower is open, upper is closed
+            expected_open_closed: (bool, bool),
+            /// expected bounds when lower is closes, upper is open
+            expected_closed_open: (bool, bool),
+            // Not: closed/closed is the same as lower/upper
+        }
+
+        let cases = vec![
```

Comment:
I was really confused about what the expected Intervals were, so I added test cases. it would be great if someone could double check these

Reply:
I think they look good according to the rules added above. But unbounded rules seem not tested?

Also an open `false` upper bound and an open `true` lower bound are not in the rules. From the test cases, looks like they are not mapped?

Reply:
> But unbounded rules seem not tested?

This is a good point, I will add tests for them